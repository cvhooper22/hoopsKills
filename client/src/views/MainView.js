import React, { useState } from 'react';
import { withRouter } from 'react-router-dom';

function MainView (props) {
  const imgSrc = './ROClogoclear.png';
  const [ id, setId ] = useState('');
  const { history } = props;

  function onSubmit(evt) {
    if (id.length) {
      console.log('onsubmit', evt);
      console.log('id is', id);
      history.push(`/qrcode/${id}`);
    }
  }
  return (
    <div className="main__container flex column ai-c">
      <img className="main__splash-image" src={ imgSrc } />
      <div className="main__instructions mt-reg">
        Here there be stats
      </div>
      <input className="main__id-input mt-reg" type="text" placeholder="Student Id" value={ id } onChange={ e => setId(e.target.value) } />
      <button className="mt-reg" type="button" onClick={ onSubmit }>Get QR Code</button>
    </div>
  );
}

export default withRouter(MainView);
