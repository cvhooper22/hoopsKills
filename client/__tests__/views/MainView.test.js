import React from 'react';
import renderer from 'react-test-renderer';
import { I18nextProvider } from 'react-i18next';

import MainView from '../../src/views/MainView';
import i18n from '../i18n';

describe('MainView -', () => {
  it('Should render MainView', () => {
    const view = renderer.create(
      <I18nextProvider i18n={ i18n }>
        <MainView
          date="2018-03-11T08:42:34"
        ></MainView>
      </I18nextProvider>,
    );

    expect(view).toMatchSnapshot();
  });
});
