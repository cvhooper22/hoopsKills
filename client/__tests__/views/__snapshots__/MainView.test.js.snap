// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainView - Should render MainView 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-md-12"
    >
      <h1>
        <i
          className="ids-icon__dashboard"
        />
         Test - Main view
      </h1>
      <div
        className="section"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Welcome to WebClient.React.Boilerplate
          </div>
        </div>
      </div>
      <div
        className="section"
      >
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Formatted time: Mar-11-2018
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Plural formatted: Test - Total 0 items
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Plural formatted: Test - Total 1 item
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Plural formatted: Test - Total 157 items
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Formatted number: 3,742
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            Test - Truncated number: 3.7k
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-md-12"
          >
            <button
              className="ids-button ids-button-normal"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span
                className="ids-button-text"
              >
                views.mainview.reloadpage
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
